CallAutomation_MediaStreaming/src/main/java/com/communication/MediaStreaming/Controllers/IncomingCallController.java

```diff
@@ -0,0 +1,96 @@
+package com.communication.MediaStreaming.Controllers;
+
+import com.communication.MediaStreaming.Logger;
+import com.azure.core.util.BinaryData;
+import com.azure.messaging.eventgrid.EventGridEvent;
+import com.azure.messaging.eventgrid.SystemEventNames;
+import com.azure.messaging.eventgrid.systemevents.SubscriptionValidationEventData;
+import com.azure.messaging.eventgrid.systemevents.SubscriptionValidationResponse;
+import com.communication.MediaStreaming.CallConfiguration;
+import com.communication.MediaStreaming.MediaStreaming;
+import com.communication.MediaStreaming.EventHandler.EventAuthHandler;
+import com.communication.MediaStreaming.EventHandler.EventDispatcher;
+
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class IncomingCallController {
+	CallConfiguration callConfiguration;
+
+	IncomingCallController(){
+		callConfiguration = CallConfiguration.initiateConfiguration();
+	}
+
+	@PostMapping(value = "/OnIncomingCall", consumes = "application/json", produces = "application/json")
+	public ResponseEntity<?> OnIncomingCall(@RequestBody(required = false) String data){
+		List<EventGridEvent> eventGridEvents = EventGridEvent.fromString(data);
+
+		if(eventGridEvents.stream().count() > 0)
+        {
+            EventGridEvent eventGridEvent = eventGridEvents.get(0);
+            BinaryData eventData = eventGridEvent.getData();
+
+            if (eventGridEvent.getEventType().equals(SystemEventNames.EVENT_GRID_SUBSCRIPTION_VALIDATION))
```

Comment:
You can change it to SystemEventNames.EVENT_GRID_SUBSCRIPTION_VALIDATION.equals(eventGridEvent.getEventType()) to avoid null pointers.
Also check if eventGridEvent could be null